src/fu-self-test.c

```diff
@@ -4946,8 +4966,18 @@ fu_plugin_composite_func(gconstpointer user_data)
 					 progress,
 					 FWUPD_INSTALL_FLAG_NONE,
 					 &error);
-	g_assert_no_error(error);
-	g_assert_true(ret);
+
+	if (scenario == 2) {
```

Comment:
can you make this an enum?  It will make it easier to follow.  Might be better to have `switch`/`case` too.

Reply:
separated the scenario in different functions